File: Recursion/python/2630.py
n = int(input())
arr = [list(map(int,input().split())) for _ in range(n)]
result = [0,0]

def search(r,c,n):
  global result
  num = arr[r][c]
  for i in range(r,r+n):
    for j in range(c,c+n):
      if (num != arr[i][j]):
        gap = n//2
        search(r,c,gap)
        search(r+gap,c,gap)
        search(r,c+gap,gap)
        search(r+gap,c+gap,gap)
        return False
  result[num] += 1
  return True

search(0,0,n)
print("\n".join(map(str,result)))

File: Recursion/javascript/2630.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const result = [0, 0];
const arr = input.slice(1, input.length - 1).map((v) => v.split(" ").map(Number));
const search = (r, c, n) => {
  const num = arr[r][c];
  for (let i = r; i < r + n; i++) {
    for (let j = c; j < c + n; j++) {
      if (num !== arr[i][j]) {
        n = ~~(n / 2);
        search(r, c, n);
        search(r + n, c, n);
        search(r, c + n, n);
        search(r + n, c + n, n);
        return false;
      }
    }
  }
  result[num]++;
  return true;
};

search(0, 0, Number(input[0]));
console.log(result.join("\n"));
